src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -71,13 +106,72 @@ public void stop() {
     spinWheels(0);
   }
 
+  public void hold() {
+    spinWheels(isConeMode ? kConeIntakeSpeed : kCubeIntakeSpeed);
+  }
+
   @Override
   public void close() throws Exception {
     // This method will close all device handles used by this object and release any other dynamic memory.
     // Mostly for JUnit tests
     motor.close();
     motor2.close();
-    
+  }
+
+  public void changeFlightState() {
+    // Check distances
+    int coneDistance = timeOfFlightSensor.getDistance0();
+    int cubeDistance = timeOfFlightSensor.getDistance1();
+
+    // Log distance values
+    coneDistLog.log((double)coneDistance);
+    cubeDistLog.log((double)cubeDistance);
+
+    // Check if sensors are online
+    boolean coneSensorUp = coneDistance != -1;
+    boolean cubeSensorUp = cubeDistance != -1;
+
+    // Log whether sensors are online
+    sensor0Up.setBoolean(coneSensorUp);
+    sensor1Up.setBoolean(cubeSensorUp);
+
+    // Log if sensors are activated
+    hasConeLog.log(coneDistance <= mmConeActivationThreshold && coneSensorUp);
```

Comment:
We don't need coneSensorUp here, because coneDistance should be -1 if the sensor is down.

Good to grab it and log it, but maybe:
coneDistance > 0 && coneDistance <=mmConeActivationThreshold?